gnucash/gtkbuilder/dialog-preferences.glade

```diff
@@ -58,6 +58,12 @@
     <property name="step_increment">0.10000000000000001</property>
     <property name="page_increment">1</property>
   </object>
+  <object class="GtkAdjustment" id="gl_number_of_days_adj">
+    <property name="lower">30</property>
+    <property name="upper">1100</property>
```

Comment:
I had thought about this and I was wondering if these settings should be saved to the .gcm file along with the other register settings which is where the GUID 000...000 is being saved. I assume this would require a feature option for the change from account file to .gcm file.
With an unfiltered GL, wouldn't that mean all transactions would be loaded!

The good thing about setting a number is that it is a rolling period so you always see 'x' amount of days. There could be another filter option to enable this for all registers with each one specifying there own limit then there would be no need for the preference option.

Now I am not sure how much to do in this PR?

Reply:
Ah, I see what you mean. The current hard-coded 30 days is converted in a dynamic starting date filter for the GL.

Without a filter would indeed mean all transactions would be loaded. I believe it's uncommon to require this, but not unthinkable. That's why it should not be the default for the General Journal, and the current 30 days sounds like a good default.

I'm not sure I understand what you mean with the change from account file to .gcm file. Do we store filter information in the account file ?

 I think allowing the user to specify a custom range would be a good thing for all registers. And I would prefer that over setting a global range for the GL only.

Reply:
The filter and sort order are stored as KVP string entries for the account like...
 <slot:key>filter</slot:key>  <slot:value type="string">0x001d,0,0</slot:value>
 <slot:key>sort-order</slot:key>  <slot:value type="string">BY_DATE_ENTERED</slot:value>

Reply:
Right. It's better to be consistent.
The minimal change would then be to store the sort order and filter in kvp as well. As the GL is not a separate account we have to find a different spot to attach this. Two possible objects can be candidates: the book account itself (there's only one GL for a book) or the root account as that one is hidden so it doesn't need filter and sort keys for itself. And in fact I like the root account best as it most closely matches the GL. If we were to display the root account with sub accounts in multi-split view it would resemble the GL pretty well.
In both cases this would modify the data format, so this particular change should go on master. And in this case I would even skip the Feature code and backwards compatibility stuff; the GL on maint won't use the kvp, but retain it if it exists. So switching between maint and master in the future will have the new option in effect in master while maint continues to behave as it does now and yet it won't interfere with master's new kvp.

Related to this I still have my reservations about storing presentation stuff in the data file directly. Or it requires a serious redesign to separate it from the data properly inside the db/xml file. It really complicates the data structure. If such a restructuring is not a feasible medium term goal we should continue to move the presentation settings to the book's metadata file (.gcm). In this context that would mean moving all the kvp data you refer to into the gcm file. And in that case we will need migration and compatibility guards in place.

Hmm, perhaps stick with the minimal effort option for now until we're ready to revisit the data/presentation separation properly...